cmd/ndm_daemonset/controller/controller.go

```diff
@@ -131,23 +130,24 @@ func NewController() (*Controller, error) {
 	}
 	controller.config = cfg
 
-	mgr, err := manager.New(controller.config, manager.Options{Namespace: Namespace})
+	// get the namespace in which NDM is installed
+	ns, err := getNamespace()
 	if err != nil {
 		return controller, err
 	}
+	controller.Namespace = ns
 
-	// Setup Scheme for all resources
-	if err := apis.AddToScheme(mgr.GetScheme()); err != nil {
+	mgr, err := manager.New(controller.config, manager.Options{Namespace: controller.Namespace})
```

Comment:
@kmova , Also here we were setting empty namespace on the manager. But since the resource had namespace field in them, it worked correctly.